src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -230,13 +229,17 @@ def get_login_credentials(self, credential_type=CredentialType.management,
         account = self.get_subscription(subscription_id)
         user_type = account[_USER_ENTITY][_USER_TYPE]
         username_or_sp_id = account[_USER_ENTITY][_USER_NAME]
+        try:
+            resource = credential_type.value
```

Comment:
For better readability, could we accept the `resource` as a parameter rather the `credential_type`. This way we don't need to detect the enum case.
If the parameter `resource is None` we can default to management